cmd/snap/complete.go

```diff
@@ -185,13 +184,18 @@ func (n assertTypeName) Complete(match string) []flags.Completion {
 type keyName string
 
 func (s keyName) Complete(match string) []flags.Completion {
-	var res []flags.Completion
-	asserts.NewGPGKeypairManager().Walk(func(_ asserts.PrivateKey, _ string, uid string) error {
-		if strings.HasPrefix(uid, match) {
-			res = append(res, flags.Completion{Item: uid})
-		}
+	keypairManager, err := getKeypairManager()
+	if err != nil {
 		return nil
-	})
+	}
+	keys, err := keypairManager.List()
+	if err != nil {
+		return nil
+	}
+	res := make([]flags.Completion, len(keys))
+	for i, k := range keys {
```

Comment:
The old NewGPGKeypairManager based code has a `strings.HasPrefix(uid, match)` check here that the new code does not have anymore. Is this no longer needed?